### DISCUSSION TITLE
请教：如何将内存中的值绑定到UI中

### DISCUSSION THREAD

**Discussion by liditingziya** [BYJRK/DotNet-Discussions#286] [C# 技术交流]
最近在给一个小游戏做汉化，游戏内的对话内容都是服务端发送的，所以我打算做一个客户端，实时获取对话内容并汉化成中文。（无法直接替换文字，游戏里的字体不支持中文）

我现在能通过CE找到对话框内容的地址，如 **GameAssembly.dll + 0xXXXX + 0xXXXX** 这样的基址，我想要在这个地址的值发生变化时能实时显示在TextBlock中，请问有什么解决方法吗（我是菜鸟 不懂hook）

**Comment by BYJRK** [comment]
https://github.com/Antelcat/I18N

这个或许能帮到你

**Reply by liditingziya** [reply]
谢谢回复！这个库我在WPF开发中已有使用过，他的动态绑定绑定的还是自己进程中的数据，而这个数据也是通过自己读取或者服务端推送而得到的。而我是想直接绑定在游戏进程中的内存地址上，而不用自己去读取内存。我现在的做法是每秒读取一次内存去拿到值并汉化，虽然勉强能够实现汉化效果，但是当出现一秒内多个对话我会漏掉内容，缩短读取间隔的话又会面临长时间没有对话浪费CPU性能。

**Comment by liditingziya** [comment]
https://github.com/erfg12/memory.dll/blob/fa6187dec24bb8c75f0cfa7882c8c8b2af4c120a/Memory/Methods/Read.cs#L431 好像也就只能自己频繁读取了
